### DISCUSSION TITLE
How can i add model in model.py ?

### DISCUSSION THREAD

**Discussion by Ekata2003** [Ekata2003/Social-media-Django#12] [Q&A] [answered]
Suggest model for user of social media platform?

**Comment by prasad-chavan1** [comment, marked as answer]
# Social Media User Model

In a Django-based social media platform, you'll typically need a user model to manage user accounts and profiles. Here's a basic user model outline:

## User Model Fields

1. **Username**: A unique field to identify users.
2. **Email**: To allow communication and account recovery.
3. **Password**: To securely store user passwords (use Django's built-in password hashing).
4. **Full Name**: For displaying the user's name on their profile.
5. **Profile Picture**: Store the URL or path to the user's profile picture.
6. **Bio**: A short user biography or description.
7. **Followers/Following**: Many-to-many relationships to represent user connections.
8. **Posts**: To link users to their created posts.

## Django Model Example

Here's a simplified representation of the user model in Django models.py:

```python
from django.db import models
from django.contrib.auth.models import AbstractUser

class User(AbstractUser):
    email = models.EmailField(unique=True)
    full_name = models.CharField(max_length=100)
    profile_picture = models.URLField(blank=True)
    bio = models.TextField(blank=True)
    followers = models.ManyToManyField('self', symmetrical=False, related_name='following')
    # Add other fields as needed.

class Post(models.Model):
    user = models.ForeignKey(User, on_delete=models.CASCADE)
    content = models.TextField()
    created_at = models.DateTimeField(auto_now_add=True)
    # Add other post-related fields.
